docs/source/concepts/ledger-model/ledger-exceptions.rst

```diff
@@ -0,0 +1,178 @@
+Rollback nodes represent a subtransaction that was cancelled and rolled
+back due to a raised exception. Rollback nodes are not considered ledger
+actions, since they have too many differences from ledger actions. As
+such, incorporating rollback nodes into the transaction structure
+requires us to redefine transactions. Transactions are now given as
+a list of actions or rollback nodes:
+
+.. code-block:: none
+
+   Transaction ::= (Action | Rollback)*
+   Rollback ::= 'Rollback' Transaction
+
+Each rollback node contains the subtransaction that was rolled back.
+
+Integrity
++++++++++
+
+The notion of a ledger action coming "after" another ledger action must
+be revised in the presence of rollback nodes. It is not enough to traverse
+the transaction tree in prefix order, because the actions under a rollback
+were rolled back.
+
+For example, a contract may be consumed by an exercise under a rollback node,
+and immediately again after the rollback node. This is allowed because the
+exercise was rolled back, and this does not represent a "double spend" of
+the same contract.
+
+We now define the "after" relation as a partial order on all the actions and
+rollback nodes of a transaction. This relation is generated by reflexivity,
+transitivity, and the following additional rules:
+
+#. If `act1` and `act2` are two actions that share a parent (e.g. two
+   top-level actions, or two actions under the same rollback node, or two
+   consequences of the same action), and `act1` precedes `act2` in the list
+   of children, then `act2` comes after `act1`.
+
+#. If `node1` is an action, and `node2` is a rollback node, and they both
+   share a parent, and `node1` precedes `node2` in the list of children,
+   then `node2` comes after `node1`.
+
+#. If `node1` and `node2` are actions or rollback nodes in the transaction
+   tree, and `node1` is a parent of `node2`, then `node2` comes after `node1`.
+
+With this modified "after" relation, the notion of consistency remains the
+same. Note that under this definition, a rollback node can be "after" its
+siblings, but never "before". The only time a rollback node comes "before"
+another node, is when that other node is a child or descendant of the
+rollback node.
+
+So in the example of a consuming exercise in a rollback node, followed by a
+consuming exercise on the same contract outside of the rollback node, neither
+exercise comes "after" the other. They are part of separate continuities,
+one of which was rolled back, and so they don't break the ledger integrity.
+
+Privacy
++++++++
+
+Rollback nodes also have an interesting effect on the notion of privacy in
+the ledger model. When projecting a transaction for a party `p`, it's
+necessary to preserve some of the rollback structure of the transaction,
+even if `p` does not have the right to observe every action under it. This
+is because we need for `p` to be able to verify that an exercise under a
```

Comment:
Well we need a bit more than that. E.g., we also need `p` to be able to verify a `Fetch` of a contract on which it's a signatory, or that a rolled-back `Create` where it was a signatory was authorized by it. I don't have an immediate suggestion for how to state this succinctly and accurately though :(